src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserHelperClasses.cs

```diff
@@ -300,103 +300,40 @@ internal void Clear()
 
     internal static class SqlEnvChangePool
     {
-        private static int _count;
-        private static int _capacity;
-        private static SqlEnvChange[] _items;
-#if DEBUG
-        private static string[] _stacks;
-#endif 
-
-        static SqlEnvChangePool()
+        private protected struct SqlEnvChangeWrapper
```

Comment:
Consider a better name for the struct.
```suggestion
        private protected struct SqlEnvChangeContainer
```

Reply:
Yup, better, changed.